boards/esp32-ethernet-kit/Makefile.dep

```diff
@@ -0,0 +1,6 @@
+include $(RIOTBOARD)/common/esp32/Makefile.dep
+
+# enables esp_eth as default network device
+ifneq (,$(filter netdev_default gnrc_netdev_default,$(USEMODULE)))
```

Comment:
```suggestion
ifneq (,$(filter netdev_default,$(USEMODULE)))
```

should be enough

Reply:
It does not compile for me if that is removed - it builds more WiFi components instead but fails to link them:

```
/home/yarrick/esp/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/bin/ld: cannot find -lcore
/home/yarrick/esp/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/bin/ld: cannot find -lrtc
/home/yarrick/esp/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/bin/ld: cannot find -lnet80211
/home/yarrick/esp/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/bin/ld: cannot find -lpp
/home/yarrick/esp/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/bin/ld: cannot find -lcoexist
/home/yarrick/esp/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/bin/ld: cannot find -lwps
/home/yarrick/esp/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/bin/ld: cannot find -lwpa
/home/yarrick/esp/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/bin/ld: cannot find -lwpa2
/home/yarrick/esp/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/bin/ld: cannot find -lphy
/home/yarrick/esp/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/bin/ld: cannot find -lespnow
/home/yarrick/esp/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/bin/ld: cannot find -lmesh
collect2: error: ld returned 1 exit status
make: *** [/home/yarrick/esp/RIOT/examples/gnrc_networking/../../Makefile.include:572: /home/yarrick/esp/RIOT/examples/gnrc_networking/bin/esp32-olimex-evb/gnrc_networking.elf] Error 1
```

It is the same for the esp32-olimex-evb board (which also has Ethernet) and esp32-wrover-kit so it is probably unrelated to my changes.

Reply:
> It is the same for the esp32-olimex-evb board (which also has Ethernet) and esp32-wrover-kit so it is probably unrelated to my changes.

In current master, it is defined exactly in that way for `esp32-olimex-evb` https://github.com/RIOT-OS/RIOT/blob/651b506fd470aaeb0517d8d6d649490db5c95180/boards/esp32-olimex-evb/Makefile.dep#L4-L6 which works wihtout any problems. There was a cleanup on May, 24th. Since module `gnrc_netdev_default` pulls in `netdev_default`, it is enough to check for the `netdev_default` module.

For the 'esp32-wrover-kit' it is different. For this board, the default Netdev is `esp_now`, which requires all these SDK WiFi libraries. But even if you pull in `esp_eth` as the default Netdev, `esp_now` is still used. Then you have two network interfaces. For this reason, the SDK WiFi components are also compiled. 

The error messages indicate that something seems to be wrong with your SDK installation. If you can compile but not link, these libraries are probably missing. Please check that these libraries are in `${ESP32_SDK_DIR}/components/esp32/lib`.

Reply:
Ah yes, I managed to get the expected version of esp-idf everywhere and now it works, showing both wired and wireless interfaces.